### PR TITLE
Feature/refactor tool search

### PR DESCRIPTION
**Pull request by Bluesmile82**
This PR refactors the search in the Sankey and map:

- Now we are using the backend search like in the homepage but filtering by context.
- Change of column and selecting a node from another column is also considered.
- Fixed some problem with the expand node on column change.

There are two cases that are still not working:

- The endpoint is returning all the years so sometimes even if we select some node it might not show in the actual year, but it will in the one that its available. We should filter the results or disable them according to the selected years

- When we select two nodes that don't have flows in common the second node won't load. 
  Try: ARAUPEL + AMINOLA. You will get the following error:
https://staging.trase.earth/api/v3/contexts/1/flows?start_year=2003&end_year=2017&include_columns=3%2C6%2C7%2C8&flow_quant=Volume&locked_nodes[]=28857&locked_nodes[]=36235&n_nodes=100&selected_nodes=28857%2C36235

In this case, we should show a banner to expand the nodes (They will then show) or to reset the Sankey options.

### CODE REVIEW

File: frontend/scripts/react-components/tool/tool-search/tool-search.saga.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { SET_SELECTED_NODES_BY_SEARCH } from 'react-components/tool/tool.actions';
+import { all, fork, takeLatest, select, put } from 'redux-saga/effects';
+import { getVisibleNodes } from 'react-components/tool/tool.selectors';
+import { expandSankey } from 'react-components/tool-links/tool-links.actions';
+
+function* checkExpandNode() {
+  function* performCheckexpandNode({ payload }) {
+    const { results } = payload;
+    const ids = results.map(n => n.id);
+    const state = yield select();
```
Comment (sorodrigo) [inline frontend/scripts/react-components/tool/tool-search/tool-search.saga.js]:
```suggestion
    const visibleNodes = yield select(getVisibleNodes);
```

File: frontend/scripts/react-components/tool/tool-search/tool-search.saga.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    const visibleNodes = getVisibleNodes(state);
```
Comment (sorodrigo) [inline frontend/scripts/react-components/tool/tool-search/tool-search.saga.js]:
```suggestion

```

**Comment by sorodrigo** [comment]
When selecting a node via search the linked geo ids should recalculate too. This can be done by subscribing the `fetchLinkedGeoIds` saga to `SET_SELECTED_NODES_BY_SEARCH`

**Comment by Bluesmile82** [comment]
Ready again

**Pull request merged**